internal/backup/volumesnapshotcontent_action.go

```diff
@@ -0,0 +1,72 @@
+/*
+Copyright 2020 the Velero contributors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package backup
+
+import (
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+
+	snapshotv1beta1api "github.com/kubernetes-csi/external-snapshotter/v2/pkg/apis/volumesnapshot/v1beta1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+
+	"github.com/vmware-tanzu/velero-plugin-for-csi/internal/util"
+	velerov1api "github.com/vmware-tanzu/velero/pkg/apis/velero/v1"
+	"github.com/vmware-tanzu/velero/pkg/plugin/velero"
+)
+
+// VolumeSnapshotContentBackupItemAction is a backup item action plugin to backup
+// CSI VolumeSnapshotcontent objects using Velero
+type VolumeSnapshotContentBackupItemAction struct {
+	Log logrus.FieldLogger
+}
+
+// AppliesTo returns information indicating that the VolumeSnapshotContentBackupItemAction action should be invoked to backup volumesnapshotcontents.
+func (p *VolumeSnapshotContentBackupItemAction) AppliesTo() (velero.ResourceSelector, error) {
+	p.Log.Info("VolumeSnapshotBackupItemAction AppliesTo")
+
+	return velero.ResourceSelector{
+		IncludedResources: []string{"volumesnapshotcontent.snapshot.storage.k8s.io"},
+	}, nil
+}
+
+// Execute backs up only those VolumeSnapshotContents that are not bound with any volumesnapshot object.
+func (p *VolumeSnapshotContentBackupItemAction) Execute(item runtime.Unstructured, backup *velerov1api.Backup) (runtime.Unstructured, []velero.ResourceIdentifier, error) {
+	p.Log.Infof("Executing VolumeSnapshotContentBackupItemAction")
+
+	var snapCont snapshotv1beta1api.VolumeSnapshotContent
+	if err := runtime.DefaultUnstructuredConverter.FromUnstructured(item.UnstructuredContent(), &snapCont); err != nil {
+		return nil, nil, errors.WithStack(err)
+	}
+
+	additionalItems := []velero.ResourceIdentifier{}
+
+	// we should backup the snapshot deletion secrets that may be referenced in the volumesnapshotcontent's annotation
+	if util.IsVolumeSnapshotContentHasDeleteSecret(&snapCont) {
+		deleteSnapshotSecretName := snapCont.Annotations[util.PrefixedSnapshotterSecretNameKey]
```

Comment:
don't really need to declare these vars, can just access the values via the `snapCont.Annotations[]` map on L65-66